src/pani/core.cljs

```diff
@@ -0,0 +1,161 @@
+(ns pani.core
+  (:refer-clojure :exclude [get-in set! reset! conj! swap! deref parents key])
+  (:require cljsjs.firebase
+            [clojure.string :as str]
+            [pani.registry :refer [register-listener]]))
+
+;; constants
+
+(def child-events
+  [:child-added
+   :child-changed
+   :child-moved
+   :child-removed])
+
+(def all-events
+  (conj child-events :value))
+
+;; utils
+
+;; FIXME: camel-case keys?
+;;        hydrate to custom vectors to preserve rich keys?
+;;        preserve sets (don't coerce to vector)
+;;        similarly preserve keywords as values
+
+(defn- hydrate [js-val]
+  (js->clj js-val :keywordize-keys true))
+
+(defn- serialize [clj-val]
+  (clj->js clj-val))
+
+(defn key
+  "Last segment in reference or snapshot path"
+  [ref]
+  (.key ref))
+
+(defn- value [snapshot]
+  (hydrate (.val snapshot)))
+
+(defn- kebab->underscore [keyword]
+  (-> keyword name (str/replace "-" "_")))
+
+(defn- underscore->kebab [string]
+  (-> string (str/replace "_" "-") keyword))
+
+(defn- korks->path [korks]
+  (if (sequential? korks)
+    (str/join "/" (map name korks))
+    (when korks (name korks))))
+
+(defn- wrap-snapshot [snapshot]
+  ;; TODO: enhance with snapshot protocol
+  [(key snapshot) (value snapshot)])
+
+;; references
+
+(defn connect
+  "Create a reference for firebase"
+  [url]
+  (js/Firebase. url))
+
+(defn get-in
+  "Obtain child reference from base by following korks"
+  [ref korks]
+  (let [path (korks->path korks)]
+    (if-not (seq path) ref (.child ref path))))
```

Comment:
Is (seq path) basically a test for `(not (blank? path))`. I'd find `blank?` easier to read in the context that `korks->path` always returns a string. Also it'd allow you to use just `if`.
Maybe just personal preference.